lib/hca/enrollment_eligibility/status_matcher.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
+module HCA
+  module EnrollmentEligibility
+    module StatusMatcher
+      module_function
+
+      CATEGORIES = [
+        {
+          enrollment_status: 'verified',
+          category: :enrolled
+        },
+        {
+          enrollment_status: ['not eligible', 'not eligible; ineligible date'],
+          text_matches: [
+            {
+              category: :inelig_not_enough_time,
+              strings: ['24 months', 'less than', '24 mos', '24months', 'two years']
+            },
+            {
+              category: :inelig_training_only,
+              strings: ['training only', 'trng only'],
+              acronyms: %w[ADT ACDUTRA ADUTRA]
+            },
+            {
+              category: :inelig_character_of_discharge,
+              strings: [
+                'other than honorable', 'dishonorable',
+                'dishonorable for va purposes', 'bad conduct', 'dis for va pur'
```

Comment:
We don't need 'dishonorable for va purposes', because 'dishonorable' covers it, unless there's some other reason to keep all of these.